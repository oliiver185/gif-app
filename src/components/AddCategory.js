import React, { useState } from 'react';
import PropTypes from 'prop-types'



const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');  //useState() asi es undefined
    
    const handleInputChange = (e) => {
        setInputValue (e.target.value);
    }

    const handleSubmit = (e) => {
          e.preventDefault(); 
          

        if (inputValue.trim().length > 2){
          setCategories(cats => [inputValue, ...cats ]);
          setInputValue('');
        }    

        // inputValue.trim().length > 2 && setCategories(cats => [...cats, inputValue]);
        // inputValue && setCategories( (category) => [...category, inputValue] ); 
    }

    return (       
        <form onSubmit={handleSubmit}>
          <p>{inputValue}</p>
        <input
          type="text"
          value = {inputValue}
          onChange= { handleInputChange}
          placeholder="Type something ðŸ˜€"
          /> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

// PrimeraApp.propTypes = {
//   saludo: PropTypes.string.isRequired
// //    otra: PropTypes.number.isRequired 
// }
export default AddCategory;