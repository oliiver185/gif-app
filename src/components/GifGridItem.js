import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url}) => {
    
    // console.log({id, title, url})
    return (
        <div className="card animate__animated animate__fadeIn">
           <img src={url} alt={title}  />

           

           <div className="hover-galeria">
               <a href={url}><p>{title}</p></a>
           </div>
           
        </div>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}


/*

1.enzyne
2. enzyme to json
3. deve de mostrar el componente correctamente 

*shallow
*wrapper
*wrapper .tomatchSnapshot()


*/