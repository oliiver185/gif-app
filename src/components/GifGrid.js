import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGif } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

      const {data: images,  loading } = useFetchGifs(category);
     
      
    // const [count, setCount] = useState(0);

    // const [images, setImages] = useState([])

    

   

    // getGif();




    return (
<>
        <h3 className="animate__animated animate__fadeIn third-title">{category}</h3>

        {loading && <p className="animate__animated animate__flash">Loading...</p>}

        {/* {loading ? 'cargando' : 'data cargada'}  */}
        <div className="card-grid">
            
            

            {/* en esta parte se puede hacer la desestructuracion de un vez sin necesidad de poner un call
            back como por ejemplo map(imagen => <li key{imagen.id>{imagen.title}</li */}

             {
               images.map(img => (
                   <GifGridItem 
                   key ={img.id}
                    {...img} /> 
                   ))
                   
            }
          
          
        </div> 

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}