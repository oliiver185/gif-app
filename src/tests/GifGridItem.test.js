import { shallow } from 'enzyme';
import React from 'react';
import { GifGridItem } from '../components/GifGridItem';



describe('should show the component correctly', () => {
    
    const title = 'hola';
    const url = 'htpps://localhost/algo.jpg'
    const wrapper = shallow(< GifGridItem title={ title } url={url} />); 
    
    test('should show the component correctly', () => {
        
            
        

            expect(wrapper).toMatchSnapshot();

    });

    test('should have a paragraph with a title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have the same image as the url and alt in props ', () => {
       
        const img = wrapper.find('img');
        // console.log(img.prop('src'));
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);


    });

    test('should have animated class', () => {
        
        const div = wrapper.find('div').at(0);

        const className = div.prop('className');


        // console.log(div.props());
        expect(className.includes('animate__fadeIn')).toBe(true)

    });
    
    
    


    
    
})
