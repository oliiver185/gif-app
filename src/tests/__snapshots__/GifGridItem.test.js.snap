// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show the component correctly should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="hola"
    src="htpps://localhost/algo.jpg"
  />
  <div
    className="hover-galeria"
  >
    <a
      href="htpps://localhost/algo.jpg"
    >
      <p>
        hola
      </p>
    </a>
  </div>
</div>
`;
